wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __WAVE_MATCHING_MATCHER_HPP__
+#define __WAVE_MATCHING_MATCHER_HPP__
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+namespace matching {
+
+// sensor data type
+template <typename R>
+class Matcher {
+  public:
+    // generic accessors
+    const Eigen::Affine3d& get_result() {return result;};
+    const Eigen::MatrixXd& get_info() {return information;};
+    const float& getRes() {return resolution;};
```

Comment:
because:
1. float is trivially small, pass by reference is good when you're passing large objects
2. you're returning a reference of `resolution` rather than value, but I'm not actually certain if `resolution` changes, it would also change the reference, e.g.

```
// suppose I do
float r;
r = Matcher.getRes();
// Matcher.resolution changes
r; // did r change as well? I'm not certain r will change as well, so why not just return a value instead of a reference
```